pkg/admission/webhook/generic_webhook.go

```diff
@@ -22,29 +22,35 @@ import (
 
 	"github.com/kcp-dev/apimachinery/pkg/logicalcluster"
 
-	"k8s.io/apimachinery/pkg/labels"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apiserver/pkg/admission"
 	webhookconfiguration "k8s.io/apiserver/pkg/admission/configuration"
 	"k8s.io/apiserver/pkg/admission/plugin/webhook"
 	"k8s.io/apiserver/pkg/admission/plugin/webhook/generic"
 	"k8s.io/apiserver/pkg/admission/plugin/webhook/rules"
 	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/client-go/tools/cache"
 	"k8s.io/klog/v2"
 
+	"github.com/kcp-dev/kcp/pkg/admission/initializers"
+	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
 	kcpinformers "github.com/kcp-dev/kcp/pkg/client/informers/externalversions"
-	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/client/listers/apis/v1alpha1"
 )
 
+const byWorkspaceIndex = "webhookDispatcher-byWorkspace"
+
+var _ initializers.WantsKcpInformers = &WebhookDispatcher{}
+
 type WebhookDispatcher struct {
-	dispatcher       generic.Dispatcher
-	hookSource       generic.Source
-	apiBinding       apisv1alpha1.APIBindingLister
-	apiBindHasSynced func() bool
+	dispatcher           generic.Dispatcher
+	hookSource           generic.Source
+	apiBindingsIndexer   cache.Indexer
+	apiBindingsHasSynced func() bool
```

Comment:
for my knowledge why the change here in the ordering of the fields?

Reply:
I only switched to the indexer instead of the lister, to avoid listing all (many!) APIBindings on a shard.